algorithms/src/sorting/Kokkos_SortByKeyPublicAPI.hpp

```diff
@@ -0,0 +1,73 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_SORT_BY_KEY_PUBLIC_API_HPP_
+#define KOKKOS_SORT_BY_KEY_PUBLIC_API_HPP_
+
+#include "./impl/Kokkos_SortByKeyImpl.hpp"
+#include <Kokkos_Core.hpp>
+#include <algorithm>
+
+namespace Kokkos {
+
+// ---------------------------------------------------------------
+// basic overloads
+// ---------------------------------------------------------------
+
+template <class ExecutionSpace, class KeysDataType, class... KeysProperties,
+          class ValuesDataType, class... ValuesProperties>
+void sort_by_key([[maybe_unused]] const ExecutionSpace& exec,
+                 Kokkos::View<KeysDataType, KeysProperties...>& keys,
+                 Kokkos::View<ValuesDataType, ValuesProperties...>& values) {
+  // constraints
+  using KeysType   = Kokkos::View<KeysDataType, KeysProperties...>;
+  using ValuesType = Kokkos::View<ValuesDataType, ValuesProperties...>;
+  static_assert(
+      KeysType::rank == 1 &&
+          (std::is_same_v<typename KeysType::array_layout, LayoutRight> ||
+           std::is_same_v<typename KeysType::array_layout, LayoutLeft> ||
+           std::is_same_v<typename KeysType::array_layout, LayoutStride>),
+      "Kokkos::sort_by_key without comparator: supports 1D keys View with "
+      "LayoutRight, LayoutLeft or LayoutStride.");
+  static_assert(
+      ValuesType::rank == 1 &&
+          (std::is_same_v<typename ValuesType::array_layout, LayoutRight> ||
+           std::is_same_v<typename ValuesType::array_layout, LayoutLeft> ||
+           std::is_same_v<typename ValuesType::array_layout, LayoutStride>),
+      "Kokkos::sort_by_key without comparator: supports 1D values View with "
+      "LayoutRight, LayoutLeft or LayoutStride.");
+
+  static_assert(SpaceAccessibility<ExecutionSpace,
+                                   typename KeysType::memory_space>::accessible,
+                "Kokkos::sort: execution space instance is not able to access "
+                "the memory space of the keys View argument!");
+  static_assert(
+      SpaceAccessibility<ExecutionSpace,
+                         typename ValuesType::memory_space>::accessible,
+      "Kokkos::sort: execution space instance is not able to access "
+      "the memory space of the values View argument!");
+
+  assert(values.extent(0) >= keys.extent(0));
+
+  if (keys.extent(0) <= 1) {
+    return;
+  }
```

Comment:
Dunno if I should repeat this crap in all `sort_by_key_<specific_implementation>` stuff. So far, I don't.